### PR TITLE
ICS27 cleanup

### PR DESCRIPTION
**Pull request by AdityaSripal**
- Minor cleanups and fixes to keep consistent with implementation

### CODE REVIEW

File: spec/app/ics-027-interchain-accounts/README.md [hunk 1]
```diff
@@ -118,10 +118,11 @@ function AuthenticateTx(msgs []Any, portId string) error {
 Executes each message sent by the owner account on the controller chain.
 
 ```typescript
-function ExecuteTx(sourcePort: string, destPort: string, destChannel: string, msgs []Any) error {
```
Comment (AdityaSripal) [inline spec/app/ics-027-interchain-accounts/README.md:121]: These fields are not used

File: spec/app/ics-027-interchain-accounts/README.md [hunk 2]
```diff
@@ -501,34 +498,29 @@ function onChanCloseConfirm(
...
-	        if err != nil {
-                    ack = NewErrorAcknowledgement(err.Error())
-                }
-	}
-
-	// NOTE: acknowledgment will be written synchronously during IBC handler execution.
-	return ack
+  switch data.Type {
+  case types.EXECUTE_TX:
+  msgs, err := types.DeserializeTx(data.Data)
```
Comment (ethanfrey) [inline spec/app/ics-027-interchain-accounts/README.md:507]: Isn't the norm "throwing exceptions" in this ics spec "pseudo code".

It seems modelled heavily on TypeScript

File: spec/app/ics-027-interchain-accounts/README.md [hunk 3]
```diff
@@ -501,34 +498,29 @@ function onChanCloseConfirm(
...
+  if err != nil {
+    return NewErrorAcknowledgement(err)
```
Comment (colin-axner) [inline spec/app/ics-027-interchain-accounts/README.md:509]: should the spec discuss the implication of returning the error in the acknowledgement? That is, including the error string in the acknowledgement makes error strings state machine breaking

Comment (colin-axner) [reply spec/app/ics-027-interchain-accounts/README.md:509]: Maybe this should be optional, as we will not include the error in the ibc-go implementation

File: spec/app/ics-027-interchain-accounts/README.md [hunk 4]
```diff
@@ -501,34 +498,29 @@ function onChanCloseConfirm(
...
+  }
+
+  // ExecuteTx calls the AuthenticateTx function defined above 
+  result, err = ExecuteTx(ctx, packet.SourcePort, packet.DestinationPort, packet.DestinationChannel, msgs)
+  if err != nil {
+    return NewErrorAcknowledgement(err)
+  }
+
+  // return acknowledgement containing the transaction result after executing on host chain
+  return NewAcknowledgement(result)
```
Comment (colin-axner) [inline spec/app/ics-027-interchain-accounts/README.md:519]: ditto for implications of including the result

**Review by seantking — APPROVED** [review]
Looks good. 

Agree with @ethanfrey about the state of the pseudo-code in general. We should have a standard for this.

File: spec/app/ics-027-interchain-accounts/README.md (hunk 2 above)
Comment (AdityaSripal) [reply spec/app/ics-027-interchain-accounts/README.md:507]: We don't want to throw an exception in this case and abort the transaction, we want to write an error acknowledgement and commit the state changes

Comment (ethanfrey) [reply spec/app/ics-027-interchain-accounts/README.md:507]: Understood. But the idiom `msg, err := fn()` is very go. The rest of the ibc specs are modelled on TypeScript.

I would say:

```ts
try {
  const msg = deserializeTx(data.data)
  // ... other stuff
} catch (err) {
  // return error packet
}
```

would look more in line with the other specs (but still needs some adjustment).

I believe the specs were intentionally not written in Go to keep a clear distinction between them and the implementation. Whether that was a good decision I will leave to you and @cwgoes to discuss. I just like consistency in the style.

File: spec/app/ics-027-interchain-accounts/README.md (hunk 3 above)
Comment (ethanfrey) [reply spec/app/ics-027-interchain-accounts/README.md:509]: > should the spec discuss the implication of returning the error in the acknowledgement? That is, including the error string in the acknowledgement makes error strings state machine breaking

Very good point.
That has proved to be the case for CosmWasm with submessages and we did hit a case where one node included the stack trace and fell out of consensus.

I like to include error message, but this is a very good thing to think through (and check code paths)

File: spec/app/ics-027-interchain-accounts/README.md (hunk 4 above)
Comment (ethanfrey) [reply spec/app/ics-027-interchain-accounts/README.md:519]: The result of a transaction is currently hashed and included in the the Tendermint block header (DataHash I believe). The data field and the code (u32) are the only fields currently consensus breaking (not events). I think including Data is fully acceptable, where errors and events may need more consideration.

Comment (colin-axner) [reply spec/app/ics-027-interchain-accounts/README.md:519]: Great insight! I did not know the data field is included in consensus. I agree, if the result is already included in consensus then it is fully acceptable to include in the acknowledgement

**Comment by AdityaSripal** [comment]
Superceded by #643

**Pull request closed without merging**